### PR TITLE
Raise fatal error If FQDN duplicated

### PR DESCRIPTION
**Pull request by linyows**
### Description

If fqdn is duplicated, it raises fatal error. Because it sometimes does not notice that it is duplicated.

### Check List

- [x] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
I tend to lean towards this being a problem with the nodes in the database with the duplicated fqdns and those need to be cleaned up.  I don't think this is the responsibility of knife ssh.

**Comment by lamont-granquist** [comment]
It should also be some long option to `knife ssh` like `--error-on-duplicated-fqdns` or something like that, which is off by default (and `--skip-on-duplicated-fqdns` would the other obvious behavior to support -- maybe `--duplicated-fqdns=(skip|error)` or something like that)

**Comment by linyows** [comment]
I think that's true. So what if you do not exit with warnings only?

**Comment by lamont-granquist** [comment]
if you're going to go down this road i think you should do what i suggested and make it a command line option, and have different behaviors - skip / error / warn and have the default be the existing behavior.

**Comment by linyows** [comment]
I understand that it should be added as such option in order not to change existing specifications.

**Comment by thommay** [comment]
Can we clarify _why_ there are duplicate fqdns here? Is this just bad data in the chef database, in which case we probably should not work around the problem. Or is there a genuine usecase?

**Comment by linyows** [comment]
I would like to solve that is not possible to execute chef-client as expected for duplicated fqdn on node files with [knife-zero](https://github.com/higanworks/knife-zero). Because knife-zero manages nodes as static file. So chef database does not matter.

File: lib/chef/knife/ssh.rb [hunk 1]
```diff
@@ -123,6 +123,11 @@ class Ssh < Knife
         :boolean => true,
         :proc => Proc.new { :raise }
 
+      option :skip_on_duplicated_fqdns,
+        :long => "--skip-on-duplicated-fqdns",
+        :description => "Skip exits even if FQDNs are duplicated",
+        :boolean => true
```
Comment (lamont-granquist) [inline lib/chef/knife/ssh.rb]: this is still the incorrect default here.  it needs to be off by default.  it should also be expanded to support different behaviors:  `:ignore`, `:warn`, `:fatal`, `:skip`

Comment (linyows) [reply lib/chef/knife/ssh.rb]: I got it.

**Comment by thommay** [comment]
@linyows this is looking good. We're gonna need you to do two things, and then we can get this merged. 
First is to fix the chefstyle failures: https://travis-ci.org/chef/chef/jobs/340760026 . You can do this by running `bundle exec chefstyle -a`.
Second is to add a DCO sign off. If you've got the chefdk installed, run `dco sign` in your branch.

**Comment by linyows** [comment]
Ok, thanks.

**Comment by thommay** [comment]
Thanks for sticking with this, and thanks for your contribution to Chef!

**Pull request merged**